src/uu/cat/src/cat.rs

```diff
@@ -32,6 +32,63 @@ mod splice;
 const USAGE: &str = help_usage!("cat.md");
 const ABOUT: &str = help_about!("cat.md");
 
+struct LineNumber {
+    buf: Vec<u8>,
+}
+
+// Logic to store a string for the line number. Manually incrementing the value
+// represented in a buffer like this is significantly faster than storing
+// a `usize` and using the standard Rust formatting macros to format a `usize`
+// to a string each time it's needed.
+// String is initialized to "     1\t" and incremented each time `increment` is
+// called. When the value overflows the range storable in the buffer, a b'1' is
+// prepended and the counting continues.
+impl LineNumber {
+    fn new() -> Self {
+        LineNumber {
+            // Initialize buf to b"     1\t"
+            buf: vec![b' ', b' ', b' ', b' ', b' ', b'1', b'\t'],
```

Comment:
Maybe you could write this as:
```suggestion
            buf: Vec::from(b"     1\t"),
```

Reply:
Thanks - I've made that change :+1: